framework/web/RequestParserInterface.php

```diff
@@ -11,15 +11,18 @@
  * Interface for classes that parse the raw request body into a parameters array.
  *
  * @author Dan Schmidt <danschmidt5189@gmail.com>
+ * @author Paul Klimov <klimov.paul@gmail.com>
  * @since 2.0
  */
 interface RequestParserInterface
 {
     /**
-     * Parses a HTTP request body.
-     * @param string $rawBody the raw HTTP request body.
-     * @param string $contentType the content type specified for the request body.
-     * @return array parameters parsed from the request body
+     * Parses a given request instance determining its body parameters.
+     * This method MUST return the array of body parameters detected from [[$request]] data.
+     * However, this method MAY adjust the given [[Request]] instance directly for extra configuration.
+     * @param Request $request the HTTP request instance to be parsed.
+     * @return array parameters parsed from the request body.
+     * @throws BadRequestHttpException in case request body format is invalid.
      */
-    public function parse($rawBody, $contentType);
+    public function parse($request);
```

Comment:
Maybe worth adding Request class type hint ?

Reply:
No, it will make it inconsistent with `ResponseFormatterInterface`:
https://github.com/yiisoft/yii2/blob/master/framework/web/ResponseFormatterInterface.php#L22